hmcts/pre-api
# S28 2882/play live event

### JIRA ticket(s)

- S28-2882

### Change description

- Adds an API endpoint which will Play a Live Event which is `Running` in AMS
- It will create a Streaming Endpoint if needed
- It will create a Streaming Locator if needed
- It will return a complete CaptureSessionDTO object

> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**

- Via the swagger UI you can create a valid API user and use this to create a court, booking and capture session
- Call the new Start Live event endpoint: PUT /media-service/live-event/start/{captureSessionId}
- Connect to the rtmps link using a CVP room
- Using this capture session id, make a call to the `PUT /media-service/streaming-locator/live-event/{captureSessionId}` endpoint
- You should receive a 200 response with capture session dto object and complete live_output_url field with capture session db record updated with status `Recording` and a populated live output url
- Error conditions:
  - Capture session id doesn't exist (404) - make up a uuid 
  - Live Event doesn't exist (404) - don't run the start live event api call first
  - Capture session in wrong state (409) - set the status of the capture session to anything but STANDBY
  - Unable to get ingest URL from AMS. No error of exception thrown (500) - don't connect the CVP room before calling  PUT /media-service/streaming-locator/live-event/{captureSessionId}
